File: Rust/src/bin/minimum_common_value.rs
fn get_common(nums1: Vec<i32>, nums2: Vec<i32>) -> i32 {
    let (mut i, mut j) = (0, 0);
    while i < nums1.len() && j < nums2.len() {
        match nums1[i].cmp(&nums2[j]) {
            std::cmp::Ordering::Equal => return nums1[i],
            std::cmp::Ordering::Less => i += 1,
            std::cmp::Ordering::Greater => j += 1,
        }
    }
    -1
}

fn main() {
    println!("{}", get_common(vec![1, 2, 3], vec![2, 4]));          // 2
    println!("{}", get_common(vec![1, 2, 3, 6], vec![2, 3, 4, 5])); // 2
    println!("{}", get_common(vec![1, 2], vec![3, 4]));             // -1
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn examples() {
        assert_eq!(get_common(vec![1, 2, 3], vec![2, 4]), 2);
        assert_eq!(get_common(vec![1, 2, 3, 6], vec![2, 3, 4, 5]), 2);
        assert_eq!(get_common(vec![1, 2], vec![3, 4]), -1);
    }
}

File: CPP/MinimumCommonValue.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int getCommon(std::vector<int>& nums1, std::vector<int>& nums2) {
        size_t i = 0, j = 0;
        while (i < nums1.size() && j < nums2.size()) {
            if (nums1[i] == nums2[j]) return nums1[i];
            if (nums1[i] < nums2[j]) ++i;
            else ++j;
        }
        return -1;
    }
};

int main() {
    Solution s;
    std::vector<int> a = {1, 2, 3}, b = {2, 4};
    std::vector<int> c = {1, 2, 3, 6}, d = {2, 3, 4, 5};
    std::vector<int> e = {1, 2}, f = {3, 4};
    std::cout << s.getCommon(a, b) << "\n"; // 2
    std::cout << s.getCommon(c, d) << "\n"; // 2
    std::cout << s.getCommon(e, f) << "\n"; // -1
    return 0;
}
